exporter/awsemfexporter/datapoint.go

```diff
@@ -213,42 +235,122 @@ func (dps histogramDataPointSlice) IsStaleNaNInf(i int) (bool, pcommon.Map) {
+// the first data point contains the first 100 buckets, while the second data point includes the remaining buckets.
+// Re-calculated Min, Max, Sum, Count for each split:
+// 1. First split datapoint:
+//   - Max: From original metric.
+//   - Min: Last bucket’s bucketBegin in the first split.
+//   - Sum: From original metric.
+//   - Count: Calculated from the first split buckets.
+//
+// 2. Second split datapoint:
+//   - Max: First bucket’s bucketEnd in the second split.
+//   - Min: From original metric.
+//   - Sum: 0.
+//   - Count: Overall count - first split count.
 func (dps exponentialHistogramDataPointSlice) CalculateDeltaDatapoints(idx int, _ string, _ bool, _ *emfCalculators) ([]dataPoint, bool) {
 	metric := dps.ExponentialHistogramDataPointSlice.At(idx)
-
 	scale := metric.Scale()
 	base := math.Pow(2, math.Pow(2, float64(-scale)))
-	arrayValues := []float64{}
-	arrayCounts := []float64{}
+
 	var bucketBegin float64
 	var bucketEnd float64
+	totalBucketLen := metric.Positive().BucketCounts().Len() + metric.Negative().BucketCounts().Len()
+	if metric.ZeroCount() > 0 {
+		totalBucketLen++
+	}
+	firstSplit := dataPointSplit{
+		values:     []float64{},
+		counts:     []float64{},
+		count:      int(metric.Count()),
+		length:     0,
+		bucketMin:  metric.Min(),
+		bucketMax:  metric.Max(),
+		maxBuckets: totalBucketLen,
+	}
+
+	currentLength := 0
+	splitThreshold := 100
+	var secondSplit dataPointSplit
+	if totalBucketLen > splitThreshold {
+		firstSplit.maxBuckets = splitThreshold
+		firstSplit.count = 0
+		secondSplit = dataPointSplit{
+			values:     []float64{},
+			counts:     []float64{},
+			count:      0,
+			length:     0,
+			bucketMin:  metric.Min(),
+			bucketMax:  metric.Max(),
+			maxBuckets: totalBucketLen - splitThreshold,
+		}
+	}
 
 	// Set mid-point of positive buckets in values/counts array.
 	positiveBuckets := metric.Positive()
 	positiveOffset := positiveBuckets.Offset()
 	positiveBucketCounts := positiveBuckets.BucketCounts()
 	bucketBegin = 0
 	bucketEnd = 0
-	for i := 0; i < positiveBucketCounts.Len(); i++ {
+	for i := positiveBucketCounts.Len() - 1; i >= 0; i-- {
 		index := i + int(positiveOffset)
-		if bucketBegin == 0 {
-			bucketBegin = math.Pow(base, float64(index))
+		if bucketEnd == 0 {
+			bucketEnd = math.Pow(base, float64(index+1))
 		} else {
-			bucketBegin = bucketEnd
+			bucketEnd = bucketBegin
 		}
-		bucketEnd = math.Pow(base, float64(index+1))
+		bucketBegin = math.Pow(base, float64(index))
 		metricVal := (bucketBegin + bucketEnd) / 2
 		count := positiveBucketCounts.At(i)
-		if count > 0 {
-			arrayValues = append(arrayValues, metricVal)
-			arrayCounts = append(arrayCounts, float64(count))
+		if count > 0 && firstSplit.length < firstSplit.maxBuckets {
+			firstSplit.values = append(firstSplit.values, metricVal)
+			firstSplit.counts = append(firstSplit.counts, float64(count))
+			firstSplit.length++
+			currentLength++
+			if firstSplit.maxBuckets < totalBucketLen {
```

Comment:
So if the total buckets is > 100, then the `firstSplit` will have a `count` of 0?

Reply:
I have updated to code to add N datapoint sequentially. each datapoint will have actual count.